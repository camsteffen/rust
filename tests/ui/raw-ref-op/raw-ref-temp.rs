// Ensure that we don't allow taking the address of temporary values

const FOUR: u64 = 4;

const PAIR: (i32, i64) = (1, 2);

const ARRAY: [i32; 2] = [1, 2];

fn main() {
    let ref_expr = &raw const 2;                                    //~ ERROR cannot take address
    let mut_ref_expr = &raw mut 3;                                  //~ ERROR cannot take address
    let ref_const = &raw const FOUR;                                //~ ERROR cannot take address
    let mut_ref_const = &raw mut FOUR;                              //~ ERROR cannot take address

    let field_ref_expr = &raw const (1, 2).0;                       //~ ERROR cannot take address
    let mut_field_ref_expr = &raw mut (1, 2).0;                     //~ ERROR cannot take address
    let field_ref = &raw const PAIR.0;                              //~ ERROR cannot take address
    let mut_field_ref = &raw mut PAIR.0;                            //~ ERROR cannot take address

    let index_ref_expr = &raw const [1, 2][0];                      //~ ERROR cannot take address
    let mut_index_ref_expr = &raw mut [1, 2][0];                    //~ ERROR cannot take address
    let index_ref = &raw const ARRAY[0];                            //~ ERROR cannot take address
    let mut_index_ref = &raw mut ARRAY[1];                          //~ ERROR cannot take address
}
