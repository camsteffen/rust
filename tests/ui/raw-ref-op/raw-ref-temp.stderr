error[E0745]: cannot take address of a temporary
  --> $DIR/raw-ref-temp.rs:10:31
   |
LL |     let ref_expr = &raw const 2;
   |                               ^ temporary value

error[E0745]: cannot take address of a temporary
  --> $DIR/raw-ref-temp.rs:11:33
   |
LL |     let mut_ref_expr = &raw mut 3;
   |                                 ^ temporary value

error[E0745]: cannot take address of a temporary
  --> $DIR/raw-ref-temp.rs:12:32
   |
LL |     let ref_const = &raw const FOUR;
   |                                ^^^^ temporary value

error[E0745]: cannot take address of a temporary
  --> $DIR/raw-ref-temp.rs:13:34
   |
LL |     let mut_ref_const = &raw mut FOUR;
   |                                  ^^^^ temporary value

error[E0745]: cannot take address of a temporary
  --> $DIR/raw-ref-temp.rs:15:37
   |
LL |     let field_ref_expr = &raw const (1, 2).0;
   |                                     ^^^^^^^^ temporary value

error[E0745]: cannot take address of a temporary
  --> $DIR/raw-ref-temp.rs:16:39
   |
LL |     let mut_field_ref_expr = &raw mut (1, 2).0;
   |                                       ^^^^^^^^ temporary value

error[E0745]: cannot take address of a temporary
  --> $DIR/raw-ref-temp.rs:17:32
   |
LL |     let field_ref = &raw const PAIR.0;
   |                                ^^^^^^ temporary value

error[E0745]: cannot take address of a temporary
  --> $DIR/raw-ref-temp.rs:18:34
   |
LL |     let mut_field_ref = &raw mut PAIR.0;
   |                                  ^^^^^^ temporary value

error[E0745]: cannot take address of a temporary
  --> $DIR/raw-ref-temp.rs:20:37
   |
LL |     let index_ref_expr = &raw const [1, 2][0];
   |                                     ^^^^^^^^^ temporary value

error[E0745]: cannot take address of a temporary
  --> $DIR/raw-ref-temp.rs:21:39
   |
LL |     let mut_index_ref_expr = &raw mut [1, 2][0];
   |                                       ^^^^^^^^^ temporary value

error[E0745]: cannot take address of a temporary
  --> $DIR/raw-ref-temp.rs:22:32
   |
LL |     let index_ref = &raw const ARRAY[0];
   |                                ^^^^^^^^ temporary value

error[E0745]: cannot take address of a temporary
  --> $DIR/raw-ref-temp.rs:23:34
   |
LL |     let mut_index_ref = &raw mut ARRAY[1];
   |                                  ^^^^^^^^ temporary value

error: aborting due to 12 previous errors

For more information about this error, try `rustc --explain E0745`.
